File: bookmark/src/main/java/com/example/api/BookmarkRestController.java
package com.example.api;

import com.example.domain.Bookmark;
import com.example.service.BookmarkService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("api/bookmarks")
public class BookmarkRestController {
    @Autowired
    BookmarkService bookmarkService;

    @RequestMapping(method = RequestMethod.GET)
    List<Bookmark> getBookmarks() {
        return bookmarkService.findAll();
    }

    @RequestMapping(method = RequestMethod.POST)
    @ResponseStatus(HttpStatus.CREATED)
    Bookmark postBookmarks(@Validated @RequestBody Bookmark bookmark) {
        return bookmarkService.save(bookmark);
    }

    @RequestMapping(value = "{id}", method = RequestMethod.DELETE)
    @ResponseStatus(HttpStatus.NO_CONTENT)
    void deleteBookmarks(@PathVariable("id") Long id) {
        bookmarkService.delete(id);
    }
}

File: bookmark-scala/src/main/scala/com/example/api/BookmarkRestController.scala
package com.example.api

import java.lang.Long

import com.example.domain.Bookmark
import com.example.service.BookmarkService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation._

@RestController
@RequestMapping(Array("api/bookmarks"))
class BookmarkRestController @Autowired()(val bookmarkService: BookmarkService) {

  @RequestMapping(method = Array(RequestMethod.GET))
  def getBookmarks = {
    bookmarkService.findAll
  }

  @RequestMapping(method = Array(RequestMethod.POST))
  @ResponseStatus(HttpStatus.CREATED)
  def postBookmarks(@Validated @RequestBody bookmark: Bookmark) = {
    bookmarkService.save(bookmark)
  }

  @RequestMapping(value = Array("{id}"), method = Array(RequestMethod.DELETE))
  @ResponseStatus(HttpStatus.NO_CONTENT)
  def deleteBookmark(@PathVariable("id") id: Long) = {
    bookmarkService.delete(id)
  }
}
